src/black/__init__.py

```diff
@@ -1847,8 +1847,11 @@ def _maybe_empty_lines_for_class_or_def(
                     newlines = 0
                 else:
                     newlines = 1
-            elif current_line.is_def and not self.previous_line.is_def:
-                # Blank line between a block of functions and a block of non-functions
+            elif (
+                current_line.is_def or current_line.is_decorator
+            ) and not self.previous_line.is_def:
```

Comment:
Right now on master Black enforces this formatting:
```pyi
class A: ...

@bar
class B: ...
    def BMethod(self) -> Any: ...

class C: ...
```

While the empty line between class B and C is right, I am not sure if the empty line between class A and class B is right since class A has an empty body. That empty line happens since Black can't tell the difference between the previous class not having an empty body or the current class being preceded by a decorator. Currently this commit keeps this behaviour since as far as I know classes don't have a reason to be decorated in stubs, but no idea whether that's right or not... 

/cc @ambv @JelleZijlstra (I barely have any experience with typing stubs so your knowledge would be appreciated)

Reply:
I would put a blank line there. Nontrivial classes (with a body that's not `...`) should have an empty line on either side.

Reply:
And there are some use cases for decorating classes in stubs, like `@runtime_checkable` for protocols.

Reply:
Sorry, I meant for the case of:
```pyi
class A: ...

@bar
class B: ...
class C: ...
```

The empty line between class A and B doesn't seem right as A and B have empty bodies. Master and this commit enforces this formatting, but I feel like it's wrong.

Reply:
Oh I see, I could go either way on that. I feel like if we do add blank lines, it should have blank lines both above and below it.